File: HackerRank/climbingTheLeaderboard.cpp
// https://www.hackerrank.com/challenges/climbing-the-leaderboard
// for dense ranking, current leaderboard given in : ranked, player's score after each game given in:player, track and display rank after every game

#include <bits/stdc.h>
using namespace std;


vector<int> climbingLeaderboard(vector<int> ranked, vector<int> player) {
	ranked.erase(unique(ranked.begin(), ranked.end()), ranked.end());
	vector<int> resultRanks;
	for (auto rank : player) {
		auto it = lower_bound(ranked.begin(), ranked.end(), rank, std::greater<int>());
		int idx = distance(ranked.begin(), it);
		resultRanks.push_back(idx + 1);
	}
	return resultRanks;
}



int main() {
	vector<int> r = { 100,90,90,80 };
	vector<int> p = { 70,80,105 };
	vector<int> res;
	res = climbingLeaderboard(r, p);
	for (auto x : res) {
		cout << x << "|";
	}
}

File: HackerRank/Other methods/climbingTheLeaderboard.py
# https://www.hackerrank.com/challenges/climbing-the-leaderboard
# for dense ranking, current leaderboard given in:ranked , player's score after each game given in:player, track and display rank after every game

def climbingLeaderboard(ranked, player):
	playerRanks=[]
	ranked_set = list(set(ranked))
	ranked_set.sort(reverse=True)
	for score in player:
		if score in ranked_set:
			print(score,"in list")
			playerRanks.append(ranked_set.index(score)+1)
		else:
			ranked_set.append(score)
			ranked_set.sort(reverse=True)
			print(score,"not in list")
			playerRanks.append(ranked_set.index(score)+1)
			ranked_set.remove(score)
	return playerRanks

r = [100,90,90,80]
p = [70,80,105]
print(climbingLeaderboard(r,p))
